### ISSUE TITLE
Causes Breaking Changes when Using Blueprint Gem

### ISSUE THREAD

**Issue by pacarvalho** [rubocop-hq/rubocop#8201]
Running `rubocop -a` causes breaking changes to https://github.com/procore/blueprinter gem serializers.

--------

## Expected behavior

Should not change blocks such as:

```ruby
  association :user, blueprint: UserSerializer do |object|
    object.author
  end
```

## Actual behavior

Changes the blocks to:

```ruby
association :user, blueprint: UserSerializer, &:author
```

## Steps to reproduce the problem

In an application that uses BluePrint for serialization and contains a serializer with block `association` or `field` params such as:

```ruby
# frozen_string_literal: true

class ReviewSerializer < ApplicationSerializer
  identifier :id

  fields :text, :created_at, :updated_at, :rating

  association :user, blueprint: UserSerializer do |object|
    object.author
  end
end
```

Run `rubocop -a`. 

## RuboCop version
0.85.1

**Comment by marcandre** [comment]
I'm not familiar with `blueprinter` but I'm curious: can you say how is that code change breaking?

**Comment by pacarvalho** [comment]
Me neither. It took me a while to narrow the bug down to that particular change. However, I have been able to reproduce reliably that it is the issue (it also happened in several other blueprint serializers). The error is the following: 

_ArgumentError: wrong number of arguments (given 1, expected 0)_

**Comment by marcandre** [comment]
Oh, right, `blueprint` is passing `object` and options (`{}`); `Symbol#to_proc` forwards all arguments.

If they did `**options` instead it would be ok (in Ruby 2.7+)...

`Style/SymbolProc` is already marked as `SafeAutoCorrect: false`. It should probably marked as `Safe: false`; it's not the autocorrection that's the issue but the diagnostic that can fail...

@pacarvalho in your source code you could use `do |object, _options|` which would be clear and silence the cop too.

**Comment by pacarvalho** [comment]
Good idea! Should I close this issue or should there be any changes made to Rubocop?

**Comment by marcandre** [comment]
Let's keep it open, I think we should mark the cop as not safe

**Comment by marcandre** [comment]
Cop was already unsafe (autocorrection), I made it unsafe (in its diagnostic).
